scripts/split-e2e-tests.ts

```diff
@@ -0,0 +1,180 @@
+import * as yaml from 'js-yaml';
+import * as glob from 'glob';
+import { join } from 'path';
+import * as fs from 'fs-extra';
+export type WorkflowJob =
+  | {
+      [name: string]: {
+        requires?: string[];
+      };
+    }
+  | string;
+
+export type CircleCIConfig = {
+  jobs: {
+    [name: string]: {
+      steps: Record<string, any>;
+    };
+  };
+  workflows: {
+    [workflowName: string]: {
+      jobs: WorkflowJob[];
+    };
+  };
+};
+
+function getTestFiles(dir: string, pattern = '**/*.test.ts'): string[] {
+  return glob.sync(pattern, { cwd: dir });
+}
+
+function generateJobName(baseName: string, testSuitePath: string): string {
+  return `${baseName}-${testSuitePath
+    .replace('src/', '')
+    .replace('__tests__/', '')
+    .replace(/\.ts$/, '')
+    .replace(/\//g, '-')
+    .replace(/\./g, '-')}`;
+}
+
+/**
+ * Takes a CircleCI config and converts each test inside that job into a separate
+ * job.
+ * @param config - CircleCI config
+ * @param jobName - job that should be split
+ * @param workflowName - workflow name where this job is run
+ * @param jobRootDir - Directory to scan for test files
+ * @param concurrency - Number of parallel jobs to run
+ */
+function splitTests(
+  config: Readonly<CircleCIConfig>,
+  jobName: string,
+  workflowName: string,
+  jobRootDir: string,
+  concurrency: number = 3,
```

Comment:
It would be nice to have the 3 in a constants with the related comments at the top of the file for easier discoverability